pkg/service/azure/volume.go

```diff
@@ -86,10 +86,30 @@ func (a *AzureController) createVolume(ctx context.Context, req *proto.CreateVol
 	if err != nil {
 		return nil, err
 	}
-	return &proto.CreateVolumeResponse{
+
+	ret := &proto.CreateVolumeResponse{
 		ResourceUri: *res.ID,
 		Volumeid:    name,
-	}, nil
+	}
+
+	if req.DeleteSnapshot {
+		//spawn a routine and let it delete
+		go func() {
+			sc, err := getSnapshotClient(cred)
+			if err != nil {
+				a.logger.Errorw("failed to get the snapshot client", "error", err)
+			}
+
+			err = a.deleteSnapshot(context.Background(), sc, cred.ResourceGroup, req.Snapshotid)
```

Comment:
Can we use `context.Todo()` instead of `context.Background()`

Reply:
Background is appropriate cintext here, 
why TODO ?

Reply:
TODO should be used when the we dont know what context we want to use, but here its not the case. We want to us background context with the routine.